.github/CONTRIBUTING.md

```diff
@@ -1,121 +1,120 @@
 # Contributing Guide
 
-* Check the [GitHub Issues](https://github.com/burnash/gspread/issues) for open issues that need attention.
-* Follow the [How to submit a contribution](https://opensource.guide/how-to-contribute/#how-to-submit-a-contribution) Guide.
+- Check the [GitHub Issues](https://github.com/burnash/gspread/issues) for open issues that need attention.
+- Follow the [How to submit a contribution](https://opensource.guide/how-to-contribute/#how-to-submit-a-contribution) Guide.
 
-* Make sure unit tests pass. Please read how to run unit tests below.
+- Make sure unit tests pass. Please read how to run unit tests [below](#run-tests-offline).
 
-* If you are fixing a bug:
-  * If you are resolving an existing issue, reference the issue id in a commit message `(fixed #XXX)`.
-  * If the issue has not been reported, please add a detailed description of the bug in the PR.
-  * Please add a regression test case.
+- If you are fixing a bug:
+  - If you are resolving an existing issue, reference the issue ID in a commit message `(e.g., fixed #XXXX)`.
+  - If the issue has not been reported, please add a detailed description of the bug in the Pull Request (PR).
+  - Please add a regression test case to check the bug is fixed.
 
-* If you are adding a new feature:
-  * Please open a suggestion issue first.
-  * Provide a convincing reason to add this feature and have it greenlighted before working on it.
-  * Add tests to cover the functionality.
+- If you are adding a new feature:
+  - Please open a suggestion issue first.
+  - Provide a convincing reason to add this feature and have it greenlighted before working on it.
+  - Add tests to cover the functionality.
 
-* Please follow [Style Guide for Python Code](https://www.python.org/dev/peps/pep-0008/).
+- Please follow [Style Guide for Python Code](https://www.python.org/dev/peps/pep-0008/).
 
-## Testing
+## CI checks
 
-1. [Obtain OAuth2 credentials from Google Developers Console](http://gspread.readthedocs.org/en/latest/oauth2.html)
+If the [test](#run-tests-offline) or [lint](#lint) commands fail, the CI will fail, and you won't be able to merge your changes into gspread.
 
-2. Run tests offline:
+Use [format](#format) to format your code before submitting a PR. Not doing so may cause [lint](#lint) to fail.
 
-Run the test suite using your current python version, in offline mode.
-This will use the curently recorded HTTP requests + responses. It does not make any HTTP call, does not require an active internet connection.
+## Install dependencies
 
-**Note:** the CI runs that command, if it fail you won't be able to merge
-your changes in GSpread.
+Installing all the `*.txt` files is optional (tox installs dependencies per-run).
 
-```
-tox -e py
+```bash
+pip install tox
+pip install -r ./test-requirements.txt -r ./lint-requirements.txt -r ./docs/requirements.txt
```

Comment:
Interestingly enough we don't require to install the dependency files. Tox will install them in a virtual env :smiley: 

The user is only *required* to installed `tox` then the rest is just useful for your editor to find the libraries installed on your computer for development purpose (mostly google dependencies)